### ISSUE TITLE
Java.lang exception

### ISSUE THREAD

**Issue by DanTUe** [utwente-fmt/vercors#126]
The following program gives a java exception stack trace:

```
class Problem {
	int[] array;
}
```

**Comment by sjcjoosten** [comment]
Thank you so much for finding this! The problem is with the ``int[] array;`` on the second line, but VerCors should of course tell you that the problem is here with a better error message!

**Comment by wytseoortwijn** [comment]
Perhaps this is clear already, but VerCors currently requires all fields of non-primitive type to be assigned some value in a constructor. For example, the following adaption would verify:

```java
class Problem {
	int[] array;
	
	Problem(int[] a) {
		array = a;
	}
}
```

The assignment `array = null;` would also be fine. So one could either improve the error messages, or simply allow such fields to be null (e.g. as a default), thereby avoiding such stack traces.

**Comment by sjcjoosten** [comment]
The option of using null by default is the one we prefer

**Comment by sjcjoosten** [comment]
Is this issue fixed as well, now that #127 is fixed?

**Comment by HenkMulder** [comment]
This issue has nothing to do with the RewriteArrayRef class. The problem is in initialization of an int[] field by an implicit constructor.

VerCors encodes an int[] field as VCTOption<Seq<Ref>>. In the implicit constructor that is generated by VerCors this field is being set to null and the value of this field is being compared with null. Which is wrong, since it should be set to, and compared to VCTNone().

**Comment by niomaster** [comment]
Fixed in #156
